database/sql/src/main/scala/cromwell/database/slick/MetadataSlickDatabase.scala

```diff
@@ -261,6 +295,38 @@ class MetadataSlickDatabase(originalDatabaseConfig: Config)
     runTransaction(action)
   }
 
+  case class SummarizationPartitionedMetadata(nonSummarizableMetadata: Seq[MetadataEntry],
+                                              summarizableRegularMetadata: Map[String, Seq[MetadataEntry]],
+                                              summarizableLabelsMetadata: Seq[MetadataEntry])
+
+  private def partitionSummarizationMetadata(rawMetadataEntries: Seq[MetadataEntry],
+                                             startMetadataKey: String,
+                                             endMetadataKey: String,
+                                             nameMetadataKey: String,
+                                             statusMetadataKey: String,
+                                             submissionMetadataKey: String,
+                                             parentWorkflowIdKey: String,
+                                             rootWorkflowIdKey: String,
+                                             labelMetadataKey: String): SummarizationPartitionedMetadata = {
+
+    val exactMatchMetadataKeys = Set(startMetadataKey, endMetadataKey, nameMetadataKey, statusMetadataKey, submissionMetadataKey, parentWorkflowIdKey, rootWorkflowIdKey)
```

Comment:
This is consistent with the existing structure in this file and package, @kshakir can explain the packaging if you like. 😉

Reply:
The way I left it, there was *only* sql-like content available in the sql packages. The `database.sql` package exposes-and-consumes only its own database classes/traits that are for the database, not the web requests, not services, etc.

For comparison, take a look at the cromiam executable, which due to "dependencies" dutifully runs its own cromwell summarizer against a forever empty, in memory, hsqldb. A waste of bits of disk (the added jar file size), memory, and cpu-- though luckily cromiam is _mostly_ idle.

Anyways, I think it helps keep a tight Separation of Concerns. It's also good practice too for coding modules, which will popup once folks start building code targeted for Java 9+. Scala will move past Java 8 compatibility someday...

As for the large number of parameters-- that's a related but different issue. I'm always of fan of [wrapping parameters into an object](https://softwareengineering.stackexchange.com/questions/145055/are-there-guidelines-on-how-many-parameters-a-function-should-accept) once one passes some "N", to be defined by one's team.

There is also always the option of passing in a function pointer, or if one wanted something more Java-esque then pass an interface/trait with a callback method. Any functions called (beyond slick) have code-smell to me, and should be warranted by a huge performance boost, such as this method.

Otherwise generally, when one is executing functions in the middle of executing sql, you're *in the middle of a transaction*, pulling data from the database, having the database wait until a JVM thread is free to process the result, then sending back more data until you release the transaction. Hopefully methods should be as quick as is technically possible, to the point of being non-existent if the business logic has done all the translation logic. Also hopefully they are run by singletons.

All of that can be overkill and taken too far though. I say go for what's needed balancing time and effort, for example leaving cromiam running an empty summarizer. 🤦